e2e/provision/playbooks/roles/bootstrap/tasks/prep-gitea.yml

```
@@ -18,6 +18,27 @@
       metadata:
         name: "{{ gitea.k8s.namespace }}"
 
+- name: Create docker hub secret
+  when:
+    - DOCKERHUB_USERNAME is defined
+    - DOCKERHUB_TOKEN is defined
+  block:
+    - name: Create docker-registry secret
+      ansible.builtin.shell: kubectl create secret docker-registry dockerhub --docker-username={{ DOCKERHUB_USERNAME }} --docker-password={{ DOCKERHUB_TOKEN }} -n {{ gitea.k8s.namespace }}
```

Shouldn't be better to create the secret resource with the k8s module?
```suggestion
  kubernetes.core.k8s:
    context: "{{ k8s.context }}"
    state: present
    definition:
      apiVersion: v1
      kind: Secret
      metadata:
        name: docker-registry
        namespace: "{{ gitea.k8s.namespace }}"
      type: Opaque
      data:
        username: {{ DOCKERHUB_USERNAME }}
        docker-password: {{ DOCKERHUB_TOKEN }}
  no_log: true
```

---

Your example uses `type: Opaque` whereas for docker hub you need `type: docker-registry`. One has to use data[.dockerconfigjson] instead. Refer https://kubernetes.io/docs/concepts/configuration/secret/#docker-config-secrets

---

Ok, so in that case you can use something like this:
```yaml
- name: Set Docker config for registry
  ansible.builtin.set_fact:
    docker_config: "{
      'auths': {
        'localhost:5000': {
          'auth': '{{ {{(lookup('ansible.builtin.env', 'DOCKERHUB_USERNAME') }} + ':' + {{(lookup('ansible.builtin.env', 'DOCKERHUB_TOKEN') }}) | b64encode }}'
        }
      }
    }"

- name: Create docker-registry secret
  kubernetes.core.k8s:
    context: "{{ k8s.context }}"
    state: present
    definition:
      apiVersion: v1
      kind: Secret
      metadata:
        name: docker-registry
        namespace: "{{ gitea.k8s.namespace }}"
      type: kubernetes.io/dockerconfigjson
      data:
        .dockerconfigjson: "{{ docker_config | to_json | b64encode }}"
no_log: true
```

---

I prefer  the command line because in the above case we are trying to construct the docker_config json which may not be well understood by all, I believe its OK to use ansible.builtin.shell with a well known documented command.
For example, when I decoded a sample .dockerconfigjson, it rendered as below whose structure looks different than above
```
{"auths":{"https://index.docker.io/v1/":{"username":"vish","password":"token","auth":"dmlzaDp0b2tlbg=="}}}
```

---

Ok, if that case, just make sure that tox functional tests is passing and don't forget to include it into the documentation

---

Is there a way that we can make tox functional tests a prow job as well?

---

I'll be more than happy to see that happen, but I'm not sure how to do it in prow, maybe @radoslawc has better idea about it. For now, I just created a [PR](https://github.com/nephio-project/test-infra/pull/86) to enable them as GitHub actions.

---

I opened a PR at https://github.com/nephio-project/docs/pull/62 for documentation update, is this what you meant?

---

No, the previous comment was referring to add tox functional test on prow